ecr-push/action.yml

```
@@ -35,10 +39,21 @@ runs:
         docker tag \
           ${{ inputs.image }}:${{ inputs.src-tag }} \
           ${{ steps.login-ecr.outputs.registry }}/${{ inputs.image }}:${digest}
-        docker tag \
-          ${{ inputs.image }}:${{ inputs.src-tag }} \
-          ${{ steps.login-ecr.outputs.registry }}/${{ inputs.image }}:${{ inputs.dest-tag }}
         docker push \
           ${{ steps.login-ecr.outputs.registry }}/${{ inputs.image }}:${digest}
-        docker push \
-          ${{ steps.login-ecr.outputs.registry }}/${{ inputs.image }}:${{ inputs.dest-tag }}
+        TAG=${{ inputs.dest-tag }}
+        while [ -n "${TAG}" ]; do
+          docker tag \
+            ${{ inputs.image }}:${{ inputs.src-tag }} \
+            ${{ steps.login-ecr.outputs.registry }}/${{ inputs.image }}:"${TAG}"
+          docker push \
+            ${{ steps.login-ecr.outputs.registry }}/${{ inputs.image }}:"${TAG}"
+          if [ ${{ inputs.push-multi-level-tags }} == "false" ]; then
+            break
+          fi
+          TRIMMED_TAG="${TAG%.*}"
```

I think it's better stop trimming if the tag contains `-` to prevent pushing `v1` on `v1.0.0-rc1`
```suggestion
          if [[ "${TAG}" == *-* ]]; then
            break
          fi
          TRIMMED_TAG="${TAG%.*}"
```